src/sparseml/evaluation/integrations/perplexity.py

```diff
@@ -0,0 +1,155 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+
+import numpy
+import torch
+from datasets import load_dataset
+from torch.nn import CrossEntropyLoss
+from tqdm import tqdm
+
+from huggingface_hub import hf_hub_download
+from sparseml.evaluation.registry import SparseMLEvaluationRegistry
+from sparseml.transformers.integration_helper_functions import create_model
+
+
+@SparseMLEvaluationRegistry.register("perplexity")
+def perplexity_eval(
+    target,
```

Comment:
why are we creating the model inside the evaluator function and not before? Seems wasteful to add model initialisation logic to every evaluator function. So I suggest `target: Module`.

Reply:
This was done because we need the target_name or model_path to create the tokenizer, which would be not be available at the integration level if we pass in the module.